heat/core/manipulations.py

```diff
@@ -609,41 +603,32 @@ def unique(a, sorted=False, return_inverse=False, axis=None):
                         inverse_indices[begin + num] = g_inverse[begin + x]
 
     else:
+        # Tensor is already split and does not need to be redistributed afterward
+        split = None
+        is_split = a.split
         max_uniques, max_pos = uniques_buf.max(0)
-
         # find indices of vectors
         if a.comm.Get_rank() == max_pos.item():
-            # Get the indices of the vectors we need from each process
-            indices = []
-            found = []
-            pos_list = inverse_pos.tolist()
-            for p in pos_list:
-                if p not in found:
-                    found += [p]
-                    indices += [pos_list.index(p)]
-                if len(indices) is max_uniques.item():
-                    break
-            indices = torch.tensor(indices, dtype=a.dtype.torch_type())
+            # Get indices of the unique vectors to share with all over processes
+            indices = inverse_pos.reshape(-1).unique()
         else:
-            indices = torch.empty((max_uniques.item(),), dtype=a.dtype.torch_type())
+            indices = torch.empty((max_uniques.item(),), dtype=inverse_pos.dtype)
 
         a.comm.Bcast(indices, root=max_pos)
+
         gres = local_data[indices.tolist()]
 
-        is_split = a.split
         inverse_indices = indices
+        if sorted:
```

Comment:
should be an issue (if it isn't already) and then this can be removed or raise/return NotImplement/Error